src/coreComponents/physicsSolvers/fluidFlow/StatOutputController.cpp

```diff
@@ -0,0 +1,170 @@
+                                                   string_view path,
+                                                   string_view fieldName,
+                                                   string_array & packCollectionPaths )
+{
+  string const taskManagerKey = GEOS_FMT( "packCollection{}{}", regionName, fieldName );
+  PackCollection * packCollection = &taskManager.registerGroup< PackCollection >( taskManagerKey );
+  string & pcObjectPath = packCollection->getReference< string >( PackCollection::viewKeysStruct::objectPathString());
+  pcObjectPath = path;
+  string & pcName = packCollection->getReference< string >( PackCollection::viewKeysStruct::fieldNameString());
+  pcName = fieldName;
+
+  m_packCollections.push_back( packCollection );
+  packCollectionPaths.emplace_back( GEOS_FMT( "{}", packCollection->getPath()) );
+}
+
+void StatOutputController::generateTimeHistory( OutputManager & outputManager,
+                                                string_view regionName,
+                                                string_array const & packCollectionPaths )
+{
+  string const outputManagerKey = GEOS_FMT( "compFlowHistory_{}", regionName );
+  //we expect timeHistory will add the extension by itself
+  makeDirsForPath( m_outputDir );
+  string const filename = GEOS_FMT( "{}/{}", m_outputDir, outputManagerKey );
+
+  TimeHistoryOutput * timeHistory = &outputManager.registerGroup< TimeHistoryOutput >( outputManagerKey );
+  string_array & collectorPaths =  timeHistory->getReference< string_array >( TimeHistoryOutput::viewKeys::timeHistoryOutputTargetString() );
+  collectorPaths = packCollectionPaths;
+  string & outputFile =  timeHistory->getReference< string >( TimeHistoryOutput::viewKeys::timeHistoryOutputFilenameString() );
+  outputFile = filename;
+
+  m_timeHistories.push_back( timeHistory );
+}
+
+StatOutputController::StatOutputController( const string & name,
+                                            Group * const parent ):
+  TaskBase( name, parent ),
+  m_statistics( nullptr )
+{
+  registerWrapper( viewKeyStruct::outputDirString(), &m_outputDir ).
+    setApplyDefaultValue( "." ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Output directory for statistics HDF regions file" );
+}
+
+void StatOutputController::initializePreSubGroups()
+{
+
+  Group & problemManager = this->getGroupByPath( "/Problem" );
+  DomainPartition & domain = problemManager.getGroup< DomainPartition >( "domain" );
+  TasksManager & taskManager = this->getGroupByPath< TasksManager >( "/Tasks" );
+  OutputManager & outputManager = this->getGroupByPath< OutputManager >( "/Outputs" );
+
+  Group & meshBodies = domain.getMeshBodies();
+  std::vector< string > const groupNames = this->getSubGroupsNames();
+
+  GEOS_ERROR_IF( groupNames.empty() || groupNames.size() >= 2,
+                 "StatOutputController must have one of the following components : {SinglePhaseStatistics, CompositionalMultiphaseStatistics} " );
```

Comment:
Not needed, check the following comment.

Reply:
The followings comment doesn't check the fact that we couldn't add 0 or more than 2 compo @MelReyCG

Reply:
needed for runtime and replace with check  `size() = 1`